### PR TITLE
Prototype defining protos for statement and predicates

### PR DESCRIPTION
**Pull request by TomHennen**
This includes an example program that uses the protos written
in Go, as well as the Go code generated from those protos
([as is standard practice for Go](https://go.dev/doc/articles/go_command#:~:text=and%20then%20check%20those%20generated%20source%20files%20into%20your%20repository)).

### CODE REVIEW

**Comment by marcelamelara** [comment]
> It could be that the best thing to do is have an automated Action that builds the code & commits it or publishes it to whatever the language repo is. We could also take the approach that we only define the protos and folks are responsible for building their own libs.

My 2 ct is that we might actually want to do both. Providing a Makefile will work in most cases, and the action for pulling and building this code in whatever language a repo is can be used for people who want to use GHA (and that action can then just use the Makefile under the hood).

**Comment by TomHennen** [comment]
Perhaps for an MVP we 

1. Create the protos.
2. Have a Makefile for building the golang libs
3. Have an example go program that shows using them

Then we leave publishing any prebuilts for another time?

**Comment by TomHennen** [comment]
Added a Makefile, still probably want to

1. Define protos for the Reference type from #133 when ready
2. Improve the test program that uses the protos (and add that use to the makefile)
3. Add something to the readme?

**Comment by TomHennen** [comment]
Ok, sorry for all the noise.

* The example program has been improved & simplified.
* After [reading this](https://go.dev/doc/articles/go_command#:~:text=and%20then%20check%20those%20generated%20source%20files%20into%20your%20repository) I decided it was best to check in the generated code so that it works with typical go workflows.
* Added proto.md to explain what's going on.
* Not adding to the readme yet since that has changes in-flight in #135 (maybe it never needs to?)
* We can postpone creating the ReferenceAttestation type until #133 is submitted?

Any thoughts?

**Review by marcelamelara — CHANGES_REQUESTED** [review]
Overall, my main point of feedback is that I'd like to have these files organized a bit more within the repo. Here's my proposed organization: (1) place all generated go files under a new `go/` directory so they're all in one place, (2) the documentation should go under `docs/`, and (3) the .protos files are fine under `spec/`.

I also think we should add a short section to the top-level README pointing to the protos.md document (with disclaimer about lack of thorough testing).

File: protos.md
Comment (TomHennen) [reply protos.md]: Done

File: spec/v1.0-draft/statement.proto
Comment (TomHennen) [reply spec/v1.0-draft/statement.proto:19]: Added a general comment at the top of each proto that the user is responsible for validating fields.  (Since we don't do validation anywhere else, not just for digest.)

File: spec/predicates/vsa.proto
Comment (TomHennen) [reply spec/predicates/vsa.proto:10]: Done, mostly a placeholder pointing at the existing SLSA doc.

**Comment by TomHennen** [comment]
Thanks @marcelamelara I've made all the changes except the readme update since I'd like to wait for #135 to be merged.

WDYT?

File: spec/predicates/vsa.proto
Comment (marcelamelara) [reply spec/predicates/vsa.proto:10]: Thanks! I think that's fine for now.

**Review by marcelamelara — APPROVED** [review]
LGTM. Making README changes after #135 makes sense to me.

**Comment by TomHennen** [comment]
I should probably squash/rebase this before we merge, so many commits...

**Review by pxp928 — APPROVED** [review]
some comments but looks good otherwise! Thanks for putting this together @TomHennen. I agree with the squash and rebase

File: spec/predicates/vsa.md
Comment (marcelamelara) [reply spec/predicates/vsa.md:1]: I've been wondering the same. Once this is merged, I'd like to start working on the SCAI protos, for example.

File: docs/protos.md
Comment (marcelamelara) [reply docs/protos.md:4]: Eventually, we'd want in-toto-golang to use these protos, too, though, I think. From that standpoint, what would be most convenient?

Comment (TomHennen) [reply docs/protos.md:4]: I can really go either way.  I actually think that the generated code _is not_ an example, and I hope people can use it directly.  We should see what @marcelamelara thinks because her intuition was that the go code should be separated.

Comment (TomHennen) [reply docs/protos.md:4]: The way it's defined here, folks (including in-toto-golang) will be able to import the generated code.  They'll just need to use the correct import path.  The difference is really if they import `"github.com/in-toto/attestation/go/spec/v1.0-draft"` or `"github.com/in-toto/attestation/spec/v1.0-draft"`.

Comment (marcelamelara) [reply docs/protos.md:4]: My main reasoning behind separating the go code was because I'm imagining having pre-generated code for other languages as well in the future (Rust, Python etc.), and having them all live with the protos seemed potentially problematic. Could we simplify the import path to something like `github.com/in-toto/attestation/go` and `github.com/in-toto/attestation/go/predicates`, respectively, or does this go against golang conventions? (I'm not a big go coder).

Comment (TomHennen) [reply docs/protos.md:4]: It might be possible and I can try once we figure out where we want the generated code to live.

For other languages I think best practices for them isn't to check the code in but just to publish to their respective distribution systems (e.g. pip).  Though I'm not sure about Rust.

Comment (TomHennen) [reply docs/protos.md:4]: I should add that I don't think having 'go' or 'spec' in the import path is that bad, and I do think it's a nice property that the code is separated.  So yeah, doesn't really matter much to me.

File: go/example/main.go
Comment (TomHennen) [reply go/example/main.go]: Done

File: go/example/main.go
Comment (TomHennen) [reply go/example/main.go]: Done

File: spec/predicates/vsa.md
Comment (TomHennen) [reply spec/predicates/vsa.md]: Moved vsa stuff into a vsa dir.

**Comment by TomHennen** [comment]
> 

Squash/rebased!

File: docs/protos.md
Comment (TomHennen) [reply docs/protos.md]: I propose that we keep this as is.  Any objections?

**Review by trishankatdatadog — APPROVED** [review]

File: docs/protos.md
Comment (marcelamelara) [reply docs/protos.md]: Nope, what we have now is fine by me.

**Pull request merged**
